modules/base/src/processors/surfaceextractionprocessor.cpp

```diff
@@ -211,16 +211,17 @@ void SurfaceExtraction::process() {
                 jobs.push_back(computeSurface(getColor(i), data));
                 inds.push_back(i);
             } else if (colors_[i]->isModified()) {
-                jobs.push_back(changeColor(getColor(i), meshes_[i]));
+                jobs.push_back(changeColor(getColor(i), (*meshes_)[i]));
                 inds.push_back(i);
             }
         }
         if (!jobs.empty()) {
             dispatchMany(jobs, [this, inds](std::vector<std::shared_ptr<Mesh>> results) {
                 for (auto [i, result] : util::zip(inds, results)) {
-                    meshes_[i] = result;
+                    auto it = meshes_->begin() + i;
+                    meshes_->replace(it, result);
```

Comment:
It would be better to output a new DataSequence each time. It can still reuse the unchanged meshes from the old one.

Reply:
We should not modify old data unless we have a very good reason to...